scripts/ci/kubernetes/minikube/start_minikube.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Guard against a kubernetes cluster already being up
+kubectl get pods &> /dev/null
```

Comment:
The assumption here is that kubectl isn't configured to talk to another cluster and that the other cluster isn't the default.  I don't think there is a better way to prove you're talking to minikube as intended, i guess you could look at the current context?

There's not a great way to parse that the config context stuff since you can't (or previously couldn't) change output types to something easily parsable.
  

Some other options here would be to ensure minikube is running using something like `minikube status` and that kubectl is configured to talk to the minikube cluster with `minikube update-context` ?

Reply:
To prevent collisions, `minikube profile` could be used to set a kubectl context name that's unique enough to be unlikely accidentally used. Then `kubectl config set-context` (or alias kubectl) could be used to ensure that minikube is used.

Reply:
@grantnicholas I think @cmkimerer and @ethernetdan have a good point here, do you want to set this up? (I can do it this weekend if you're busy)

Reply:
Drive-by thought - Isn't this meant to run under travis only? In that case, there is complete control over the environment and we can be sure there isn't a different k8s cluster configured also. 

For the cleanup, we usually run the tests in a separate randomly generated namespace every time and clean up the namespace which ensures everything - i.e. secrets, services, pods, etc get cleaned up.

Reply:
fair point. The other consideration is making it as easy as possible for users to unit test if they want to add features, but even then there's no reason they HAVE to use minikube (at least none of our unit tests are minikube specific).

Reply:
Given the comment here: https://github.com/apache/incubator-airflow/pull/2853/files#diff-a6fe26ef61545b07f21f31c42a69d4e3R22, it looks like the desired way to do development work on this also calls this script.

However, if you had just started minikube though as that linked documentation also suggests, you also have likely just re-configured your kubectl's context.

It does seem like the `setup_kubernetes.sh` script isn't all that helpful in setting up the development environment as it is now though.  Maybe it was intended to be more in-depth, or was at one point and lost (unneeded) functionality as this patch evolved?

Reply:
Yeah, the goal is also to support user-run testing, we could separate out the actual testing from the setup into a script which takes `master` as a parameter. Not sure if that's worth the effort right now.

Reply:
Eh, I'm fine to put that on the roadmap, but I really want to get people trying this out as an MVP/alpha

Reply:
@cmkimerer While the script is technically only supposed to be used for travis, it's nice to have a way to setup a "standard" development environment. I know when we were all collaborating it was hard to prevent "works on my machine" issues with different kubernetes setups so having one standard setup enforced a certain amount of reliability. 

I think we probably want to make this a bit more robust, but probably not a blocker for a first release (imo).

Reply:
@grantnicholas Yea, I'm definitely not trying to hold up this from landing (as I'm waiting on this and the executor PR 😄 to land), just figured I'd call out the fact this is potentially dangerous and is something we should smooth out [in a future release]